Grammar/python.gram

```diff
@@ -475,6 +475,7 @@ await_primary[expr_ty] (memo):
     | AWAIT a=primary { CHECK_VERSION(expr_ty, 5, "Await expressions are", _Py_Await(a, EXTRA)) }
     | primary
 primary[expr_ty]:
+    | invalid_primary
```

Comment:
I suppose there's a reason the error alternative comes first? Is it because this rule is so recursive?

Reply:
It doesn't have to be first, but it does have to be before the `primary genexp` alternative. There's a thorough explanation of why this is on the issue. Should I add a comment about that in the grammar?

Reply:
Maybe keep it where it is and add a brief comment "must be before 'primay genexp' because of invalid_genexp"?

Reply:
Done in https://github.com/python/cpython/pull/22996/commits/aada2f0abe0a823366e9a33b2826d83b1c746e0f.